x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_nav.test.tsx

```diff
@@ -82,106 +82,153 @@ describe('EngineRouter', () => {
+    expect(wrapper.find('[data-test-subj="EngineNavSchemaUnconfirmedFields"]')).toHaveLength(1);
 
-    // TODO: Schema warning icon
+    setMockValues({ ...values, myRole, hasSchemaConflicts: true });
+    wrapper.setProps({}); // Re-render
+    expect(wrapper.find('[data-test-subj="EngineNavSchemaConflicts"]')).toHaveLength(1);
   });
 
-  // TODO: Unskip when EngineLogic is migrated
-  it.skip('renders a crawler link', () => {
-    setMockValues({ myRole: { canViewEngineCrawler: true } });
+  it('renders a crawler link', () => {
+    const myRole = { canViewEngineCrawler: true };
+    setMockValues({ ...values, myRole });
     const wrapper = shallow(<EngineNav />);
     expect(wrapper.find('[data-test-subj="EngineCrawlerLink"]')).toHaveLength(1);
 
-    // TODO: Test that the crawler link does NOT show up for meta/sample engines
+    // Does not render for meta engines
+    setMockValues({ ...values, myRole, isMetaEngine: true });
+    wrapper.setProps({}); // Re-render
+    expect(wrapper.find('[data-test-subj="EngineCrawlerLink"]')).toHaveLength(0);
+
+    // Does not render for sample engine
+    setMockValues({ ...values, myRole, isSampleEngine: true });
+    wrapper.setProps({}); // Re-render
+    expect(wrapper.find('[data-test-subj="EngineCrawlerLink"]')).toHaveLength(0);
   });
 
-  // TODO: Unskip when EngineLogic is migrated
-  it.skip('renders a meta engine source engines link', () => {
-    setMockValues({ myRole: { canViewMetaEngineSourceEngines: true } });
+  it('renders a meta engine source engines link', () => {
+    const myRole = { canViewMetaEngineSourceEngines: true };
+    setMockValues({ ...values, myRole, isMetaEngine: true });
     const wrapper = shallow(<EngineNav />);
     expect(wrapper.find('[data-test-subj="MetaEngineEnginesLink"]')).toHaveLength(1);
 
-    // TODO: Test that the crawler link does NOT show up for non-meta engines
+    // Does not render if engine is not a meta-engine
+    setMockValues({ ...values, myRole, isMetaEngine: false });
+    wrapper.setProps({}); // Re-render
+    expect(wrapper.find('[data-test-subj="MetaEngineEnginesLink"]')).toHaveLength(0);
   });
 
   it('renders a relevance tuning link', () => {
-    setMockValues({ myRole: { canManageEngineRelevanceTuning: true } });
+    setMockValues({ ...values, myRole: { canManageEngineRelevanceTuning: true } });
     const wrapper = shallow(<EngineNav />);
     expect(wrapper.find('[data-test-subj="EngineRelevanceTuningLink"]')).toHaveLength(1);
+  });
+
+  it('renders relevance tuning nav icons', () => {
+    setMockValues({
+      ...values,
+      myRole: { canManageEngineRelevanceTuning: true },
+      engine: {
+        unsearchedUnconfirmedFields: true,
+        invalidBoosts: true,
+      },
+    });
+    const wrapper = shallow(<EngineNav />);
 
-    // TODO: Boost error icon
+    expect(wrapper.find('[data-test-subj="EngineNavRelevanceTuningInvalidBoosts"]')).toHaveLength(1); // prettier-ignore
```

Comment:
Why did you add the prettier-ignore comments here? Even if it is not optimal, I prefer that to placing these ignores all over the place.

Reply:
It does this without it :dead_inside:

```
    expect(wrapper.find('[data-test-subj="EngineNavRelevanceTuningInvalidBoosts"]')).toHaveLength(
      1
    );
    expect(
      wrapper.find('[data-test-subj="EngineNavRelevanceTuningUnsearchedFields"]')
    ).toHaveLength(1);
```

Which is some... real... 'what the fuck prettier', especially on that first line. lol. It doesn't look so bad with the tests split up now though, so I'll just... leave it I guess 😐

Reply:
https://github.com/elastic/kibana/pull/83138/commits/e10cc76df71ad4845d3f54e925a31bbb326f6ea9

Reply:
lol, yikes